pkg/cmd/codespace/create.go

```diff
@@ -111,6 +115,53 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 		branch = repository.DefaultBranch
 	}
 
+	devContainerPath := opts.devContainerPath
+
+	// now that we have repo+branch, we can list available devcontainer.json files (if any)
+	if len(opts.devContainerPath) < 1 {
+		a.StartProgressIndicatorWithLabel("Fetching devcontainer.json files")
+		devcontainers, err := a.apiClient.ListDevContainers(ctx, repository.ID, branch, 100)
+		if err != nil {
+			return fmt.Errorf("error getting devcontainer.json paths: %w", err)
+		}
+		a.StopProgressIndicator()
+
+		if len(devcontainers) > 0 {
+
+			// if there is only one devcontainer.json file and it is one of the default paths we can auto-select it
+			if len(devcontainers) == 1 && utils.StringInSlice(devcontainers[0].Path, DEFAULT_DEVCONTAINER_DEFINITIONS) {
```

Comment:
And if there is a single `devcontainer.json` file but it isn't one of the default paths, would it be not safe to auto-select it? Is this because the generic `Default Codespaces configuration` option needs to be added as an option as well?

Reply:
This has been a hotly debated product question and they've landed on trying to make sure users are never surprised by a configuration. The rationale is that anyone with write access to the repo could be the one to introduce such a devcontainer.json and anyone else who creates a codespace from that repo would be silently taking on that configuration if we auto-selected it for them. We only auto-select when the pre-multidevcontainer behavior would have auto-selected for you (one of the default paths).